syntax = "proto3";

message File{
	string file_name = 1;
}

message Credentials {
	string username = 1;
	string password = 2;
}

message Url {
	string url = 1;
}

message Reply{
	int32 status_code = 1;
}

message Index {
	repeated string my_list = 1; 
}

message UploadMessage{
	string url = 1;
	string file_name = 2;
}

message Any{

}

service PServer{
    rpc DownloadFile(File) returns (Reply) {}
	rpc UploadFile(File) returns (Reply) {}

	rpc RequestFile(UploadMessage) returns (Reply) {}
	rpc RequestUpload(File) returns (Reply) {}

	rpc RequestLogIn(Credentials) returns (Reply) {}
	rpc RequestLogOut(Url) returns (Reply) {}

	rpc RequestPinging(Any) returns (Reply) {}

	rpc ListIndex(Any) returns (Index) {}
}